src/test/java/io/sealights/plugins/sealightsjenkins/EndExecutionPostBuildStepTest.java

```diff
@@ -0,0 +1,110 @@
+import io.sealights.agents.infra.integration.enums.LogLevel;
+import io.sealights.plugins.sealightsjenkins.buildsteps.cli.CLIHandler;
+import io.sealights.plugins.sealightsjenkins.buildsteps.cli.CommandMode;
+import io.sealights.plugins.sealightsjenkins.buildsteps.cli.LogConfiguration;
+import io.sealights.plugins.sealightsjenkins.utils.Logger;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.io.PrintStream;
+
+import static org.mockito.Mockito.*;
+
+public class EndExecutionPostBuildStepTest {
+    private CLIHandler cliHandler;
+    private AbstractBuild<?, ?> build;
+    private EnvVars envVars;
+    private Launcher launcher;
+    private BuildListener listener;
+    private PrintStream logger;
+
+    @Before
+    public void setUp() throws Exception {
+        cliHandler = mock(CLIHandler.class);
+        build = mock(AbstractBuild.class);
+        envVars = spy(new EnvVars());
+        launcher = mock(Launcher.class);
+        listener = mock(BuildListener.class);
+        logger = mock(PrintStream.class);
+        when(cliHandler.handle()).thenReturn(true);
+        when(build.getEnvironment(any(BuildListener.class))).thenReturn(envVars);
+        when(listener.getLogger()).thenReturn(logger);
+    }
+
+    @Test
+    public void perform_withLabIdFromAdditionalArgs_shouldUseIt() throws IOException, InterruptedException {
+        String lab1 = "lab1";
+        EndExecutionPostBuildStep endExecutionPostBuildStep = createInstance("bsid", "labid=" + lab1);
+        EndExecutionPostBuildStep spy = spy(endExecutionPostBuildStep);
+
+        spy.perform(build, launcher, listener);
+
+        verify(spy).runCLICommand(eq(build), eq(launcher), eq(listener), any(CommandMode.EndView.class),
+                any(Logger.class), any(LogConfiguration.class), eq(lab1));
+    }
+
+    @Test
+    public void perform_bsidNotResolved_shouldNotRunCommand() throws IOException, InterruptedException {
+        EndExecutionPostBuildStep endExecutionPostBuildStep = createInstance(EndExecutionPostBuildStep.DEFAULT_BSID_VAL,
+         null);
+        EndExecutionPostBuildStep spy = spy(endExecutionPostBuildStep);
+
+        spy.perform(build, launcher, listener);
+
+        verify(spy, never()).runCLICommand(eq(build), eq(launcher), eq(listener), any(CommandMode.EndView.class),
+                any(Logger.class), any(LogConfiguration.class), eq(""));
+    }
+
+    @Test
+    public void perform_bsidNull_shouldNotRunCommand() throws IOException, InterruptedException {
+        EndExecutionPostBuildStep endExecutionPostBuildStep = createInstance(null,
+                null);
+        EndExecutionPostBuildStep spy = spy(endExecutionPostBuildStep);
+
+        spy.perform(build, launcher, listener);
+
+        verify(spy, never()).runCLICommand(eq(build), eq(launcher), eq(listener), any(CommandMode.EndView.class),
+                any(Logger.class), any(LogConfiguration.class), eq(""));
```

Comment:
Is there a reason not to use `anyString()` in `perform_bsidNull_shouldNotRunCommand` and `perform_bsidNotResolved_shouldNotRunCommand` as the lab id?

Reply:
No preticular reason, I'm agree that `anyString()` is more correct